tests/integration/s3/test_s3.py

```diff
@@ -2345,6 +2345,22 @@ def test_range_header_body_length(self, s3_bucket, snapshot, aws_client):
         assert chunk_size == len(content)
         snapshot.match("get-object", resp)
 
+    @pytest.mark.only_localstack
+    def test_download_fileobj_multiple_range_requests(self, s3_bucket, snapshot, aws_client):
```

Comment:
It would be great if we could have some more streaming tests to verify that the chunks really are handled one by one.

Reply:
How do you mean? Making a manual iteration of the different chunk ranges? We have some tests with the `Range` headers already, only that one was missing firing up multiple threads to download them all at once. What do you think should be added?

Reply:
I was just thinking that there isn't really a test which would fail when the streaming (handling the chunks one-by-one while processing instead of bulk-loading them in memory before processing) is not working.
Since you introduced a parallel CI test step which executes the tests for this new opt-in provider, you could add a test like this to make sure that the streaming doesn't break (aka no component reads the whole body and with that breaks the streaming). But tests like this are also quite hard to write to be honest...

Reply:
You are right, this is pretty complicated. Maybe we can wait with a proper test until the next iteration? Then it should become way easier to test this reliably?